pkg/kubelet/api/v1alpha1/runtime/api.proto

```diff
@@ -292,77 +292,78 @@ message PodSandboxConfig {
     //
     map<string, string> annotations = 7;
     // Optional configurations specific to Linux hosts.
-    optional LinuxPodSandboxConfig linux = 8;
+    LinuxPodSandboxConfig linux = 8;
 }
 
 message RunPodSandboxRequest {
     // Configuration for creating a PodSandbox.
-    optional PodSandboxConfig config = 1;
+    PodSandboxConfig config = 1;
 }
 
 message RunPodSandboxResponse {
     // ID of the PodSandbox to run.
-    optional string pod_sandbox_id = 1;
+    string pod_sandbox_id = 1;
 }
 
 message StopPodSandboxRequest {
     // ID of the PodSandbox to stop.
-    optional string pod_sandbox_id = 1;
+    string pod_sandbox_id = 1;
 }
 
 message StopPodSandboxResponse {}
 
 message RemovePodSandboxRequest {
     // ID of the PodSandbox to remove.
-    optional string pod_sandbox_id = 1;
+    string pod_sandbox_id = 1;
 }
 
 message RemovePodSandboxResponse {}
 
 message PodSandboxStatusRequest {
     // ID of the PodSandbox for which to retrieve status.
-    optional string pod_sandbox_id = 1;
+    string pod_sandbox_id = 1;
 }
 
 // PodSandboxNetworkStatus is the status of the network for a PodSandbox.
 message PodSandboxNetworkStatus {
     // IP address of the PodSandbox.
-    optional string ip = 1;
+    string ip = 1;
 }
 
 // Namespace contains paths to the namespaces.
 message Namespace {
     // Path to the network namespace.
-    optional string network = 1;
+    string network = 1;
     // Namespace options for Linux namespaces.
-    optional NamespaceOption options = 2;
+    NamespaceOption options = 2;
 }
 
 // LinuxSandboxStatus contains status specific to Linux sandboxes.
 message LinuxPodSandboxStatus {
     // Paths to the sandbox's namespaces.
-    optional Namespace namespaces = 1;
+    Namespace namespaces = 1;
 }
 
 enum PodSandboxState {
-    SANDBOX_READY = 0;
-    SANDBOX_NOTREADY = 1;
+    SANDBOX_NOTSPECIFIED = 0;
```

Comment:
Actually I don't feel this is really necessary, the runtime should specify the state anyway before returning the status, and kubelet should be able to make assumption about that, right? @yujuhong @feiskyer